pkg/cache/inmemory.go

```diff
@@ -299,7 +342,32 @@ func (c *InMemoryCache) Fetch(ctx context.Context, keys []string) map[string][]b
 	for _, key := range keys {
 		if b, ok := c.get(key); ok {
 			results[key] = b
+		} else if c.singleFlight {
```

Comment:
I would suggest avoiding else and return above so we can have more readable code. See https://github.com/thanos-io/thanos/blob/main/docs/contributing/coding-style-guide.md#control-structure-prefer-early-returns-and-avoid-else